### PR TITLE
Consensus caller based on nh13's original branch plus joint refactoring

### PR DESCRIPTION
**Pull request by tfenne**

### CODE REVIEW

**Comment by tfenne** [comment]
@nh13 I've squashed this down into four commits that maintain a pretty good track of authorship for this stuff.  I think this is now ready for final review and merging.

File: build.sbt [hunk 1]
```diff
@@ -124,6 +124,7 @@ lazy val root = Project(id="fgbio", base=file("."))
 	    "com.fulcrumgenomics"       %% "dagr-commons"  % "0.1.1-SNAPSHOT",
 	    "com.fulcrumgenomics"       %% "dagr-sopt"     % "0.1.1-SNAPSHOT",
       "com.github.samtools"       %  "htsjdk"        % "2.5.0" excludeAll(htsjdkExcludes: _*),
+      "net.jafama"                %  "jafama"        % "2.1.0",
```
Comment (nh13) [inline build.sbt]: reminder: lets clean up the alignment above in some future commit.

File: src/main/scala/com/fulcrumgenomics/FgBioDef.scala [hunk 2]
```diff
@@ -30,4 +30,5 @@ import dagr.commons.CommonsDef
   * Place to put common function, type and implicit definitions that can be
   * imported into other classes easily.
   */
-object FgBioDef extends CommonsDef 
+object FgBioDef extends CommonsDef {
+}
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/FgBioDef.scala]: This probably can be reverted.

File: src/main/scala/com/fulcrumgenomics/util/NumericTypes.scala [hunk 3]
```diff
@@ -140,10 +138,13 @@ object NumericTypes {
...
 
     /** Computes the probability of the given independent events co-occurring: Pr(AB..N) = Pr(A)*Pr(B)*...*Pr(N). */
-    def andAll(values: Array[Double]): LogProbability = values.sum
+    def and(values: Array[Double]): LogProbability = values.sum
 
-    /** Computes the probability Pr(A and not B) = Pr(A) - Pr(B). */
-    def notOther(a: LogProbability, b: LogProbability): LogProbability = {
+    /**
+      * Computes the probability Pr(A OR not B) = Pr(A) - Pr(B). While this could be computed using
+      * or(a, not(b), this form is more efficient and more precise.
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/util/NumericTypes.scala]: `or(a, not(b)` -> `or(a, not(b))`

**Comment by nh13** [comment]
@tfenne those last few comments can likely be added to your last commit, care to do that and then we can merge?

**Pull request merged**